File: StudentsProject/front/src/app/model/Student.ts
import { Exam } from './Exam';

export interface StudentInterface {
    id?: number;
    firstName: string;
    lastName: string;
    cardNumber: string;
    exams?: Exam[];
}

export class Student implements StudentInterface {
    public id: number;
    public firstName: string;
    public lastName: string;
    public cardNumber: string;
    public exams: Exam[];

    constructor(studentCfg: StudentInterface) {
        this.id = studentCfg.id;
        this.firstName = studentCfg.firstName;
        this.lastName = studentCfg.lastName;
        this.cardNumber = studentCfg.cardNumber;
        this.exams = studentCfg.exams || [];
    }
}

File: StudentsProject/back/src/main/java/vp/advancedjava/students/model/Student.java
package vp.advancedjava.students.model;

import java.util.HashSet;
import java.util.Set;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.OneToMany;

@Entity
public class Student {
    @Id
    @GeneratedValue
    private Long id;

    private String firstName;
    private String lastName;

    @Column(nullable = false)
    private String cardNumber;

    @OneToMany(mappedBy = "student", fetch = FetchType.LAZY, cascade = CascadeType.REFRESH)
    private Set<Exam> exams = new HashSet<>();

    public Student() {

    }

    public Student(Long id, String firstName, String lastName,
                   String cardNumber) {
        super();
        this.id = id;
        this.firstName = firstName;
        this.lastName = lastName;
        this.cardNumber = cardNumber;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getCardNumber() {
        return cardNumber;
    }

    public void setCardNumber(String cardNumber) {
        this.cardNumber = cardNumber;
    }

    public Set<Exam> getExams() {
        return exams;
    }

    public void setExams(Set<Exam> exams) {
        this.exams = exams;
    }

    @Override
    public String toString() {
        return "Student [id=" + id + ", firstName=" + firstName + ", lastName="
                + lastName + ", cardNumber=" + cardNumber + "]";
    }
}
